app/models/amusement_park.rb

```diff
@@ -1,3 +1,16 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+  has_many :mechanics, through: :rides
+
+  def unique_mechanics
+    mechanics.distinct
+  end
+
+  def rides_with_mech_exp 
+    rides
+      .joins(:mechanics)
+      .select('rides.name, avg(mechanics.years_experience) as avg_exp')
+      .group('rides.name')
```

Comment:
I think grouping by the name works, you could also group by the `rides.id`.  Great job with this query.